### PR TITLE
DMP-3967 : Add some tests to verify event to hearings results

### PR DESCRIPTION
**Pull request by mestebanez**
### JIRA link (if applicable) ###

[DMP-3967](https://tools.hmcts.net/jira/browse/DMP-3967)

### Change description ###

Add some additional integartion tests to prove we only receive the event hearings that are applicable to both the case number/courthouse and case id

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/darts/cases/controller/CaseControllerGetEventByCaseIdTest.java [hunk 1]
```diff
@@ -101,6 +114,8 @@ void casesGetEventsEndpoint() throws Exception {
         expectedJson = expectedJson.replace("<hearing-id>", hearingEntity.getId().toString());
         JSONAssert.assertEquals(expectedJson, actualJson, JSONCompareMode.NON_EXTENSIBLE);
 
+        // assert that we only ever got one event. The one that was associated to the first case hearing
+        Assertions.assertEquals(1, eventEntityList.size());
```
Comment (Ben-Edwards-cgi) [inline src/integrationTest/java/uk/gov/hmcts/darts/cases/controller/CaseControllerGetEventByCaseIdTest.java:118]: We should also verify we get the correct event returned.

Comment (mestebanez) [reply src/integrationTest/java/uk/gov/hmcts/darts/cases/controller/CaseControllerGetEventByCaseIdTest.java:118]: Donot we already do that on line 113?

File: src/integrationTest/java/uk/gov/hmcts/darts/event/controller/EventsControllerCourtLogsTest.java [hunk 2]
```diff
@@ -176,6 +176,14 @@ void courtLogsGetResultMatch() throws Exception {
         var event = createEventWith(LOG, "test", hearingEntity, createOffsetDateTime("2023-07-01T10:00:00"));
         dartsDatabase.save(event);
 
+        // generate a hearing that we do not expect to be returned
```
Comment (jackmaloney) [inline src/integrationTest/java/uk/gov/hmcts/darts/event/controller/EventsControllerCourtLogsTest.java:179]: Is it worth adding abit more detail around what we are trying to achieve here? Or just link it back to the spike Jira ticket?

File: src/integrationTest/java/uk/gov/hmcts/darts/cases/controller/CaseControllerGetEventByCaseIdTest.java (hunk 1 above)
Comment (jackmaloney) [reply src/integrationTest/java/uk/gov/hmcts/darts/cases/controller/CaseControllerGetEventByCaseIdTest.java:118]: Yeh, I think this is fine.

**Review by jackmaloney — APPROVED** [review]

File: src/integrationTest/java/uk/gov/hmcts/darts/event/controller/EventsControllerCourtLogsTest.java (hunk 2 above)
Comment (mestebanez) [reply src/integrationTest/java/uk/gov/hmcts/darts/event/controller/EventsControllerCourtLogsTest.java]: Fixed

**Pull request merged**